File: 2016-maximum-difference-between-increasing-elements/2016-maximum-difference-between-increasing-elements.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const maximumDifference = nums => {
    let maxDiff = -1;
    let min = Infinity;
    for (let i = 0; i < nums.length; i++) {
        const curr = nums[i];
        min = Math.min(curr, min);
        if (curr > min) maxDiff = Math.max(maxDiff, curr - min);
    }
    return maxDiff;
};

File: 2016-maximum-difference-between-increasing-elements/2016-maximum-difference-between-increasing-elements.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        minNum = math.inf
        maxDiff = 0
        for i in range(len(nums)):
            minNum = min(minNum, nums[i])
            maxDiff = max(maxDiff, nums[i] - minNum)
        if maxDiff == 0:
            return -1
        else:
            return maxDiff